src/io/flutter/run/daemon/DeviceService.java

```diff
@@ -230,9 +232,12 @@ private DeviceDaemon chooseNextDaemon(Refreshable.Request<DeviceDaemon> request)
   public void restart() {
     if (project.isDisposed()) return;
 
-    deviceDaemon.refresh(this::shutDownDaemon);
+    JobScheduler.getScheduler().schedule(this::shutDown, 0, TimeUnit.SECONDS);
+    JobScheduler.getScheduler().schedule(this::refreshDeviceDaemon, 4, TimeUnit.SECONDS);
```

Comment:
This mystery delay allows the daemon to shutdown and restart successfully. Without this it would shutdown, but wouldn't start back up. I'm not sure if this is due to a race condition in the processes, or some bug in (or misuse of) the `Refreshable` class.

:(